python_basics_loops_conditionals/python_basics_loops_conditionals.md

```diff
@@ -0,0 +1,397 @@
+
+</div>
+
+
+### Range
+The built-in function `range()` is frequently used in loops. The `range()` function returns a range object.
+
+@sage
+<div class="python">
+<lia-keep>
+<script type="text/x-sage">
+
+print(range(1, 20))
+
+</script>
+</lia-keep>
+</div>
+
+This is not terribly interesting on its own, but is very useful if you want to loop through a sequence of numbers, such as 1 to 20, without having to manually build a list of all of those numbers. If you create a list from the range object and then print that list, you'll see what's going on behind the scenes. What do you think this code will do? Run the code cell below and find out.
+
+@sage
+<div class="python">
+<lia-keep>
+<script type="text/x-sage">
+
+print(list(range(1, 20)))
+
+</script>
+</lia-keep>
+</div>
+
+Notice that the last number listed is 19, not 20. Just like in subsetting, the first number passed to `range()` is **inclusive**, and the second is **exclusive**. And don't forget that the first position is index 0!
+
+**Your Turn**: Before running the code cell below, try to predict what number will be returned. Got a number in mind? Run the code and see if your hypothesis is correct! (**Hint**: the bracket notation below (`[ ]`) allows Python to access an element in a collection object, like a list; you saw it before when indexing and subsetting).
+
+@sage
+<div class="python">
+<lia-keep>
+<script type="text/x-sage">
+
+print(list(range(1, 20))[7])
+
+</script>
+</lia-keep>
+</div>
+
+
+### Quiz: Loops
+
+
+@sage
+<div class="python">
+<lia-keep>
+<script type="text/x-sage">
+
+for i in range(0, 6):
+  j = i*i
+</script>
+</lia-keep>
+</div>
+
+The loop in the code cell above is missing a `print()` statement to show us the output. How would you include a `print()` statement such that all of the squares of `i` in `range(0,6)` are displayed? Feel free to edit the code cell above and experiment to find the answer.
+
+[( )] `print(i)`, indented so that it **inside** the loop
+[(X)] `print(j)`, indented so that it **inside** the loop
+[( )] `print(i)` with no indentation, so that it is **outside** of the loop
+[( )] `print(j)` with no indentation, so that it is **outside** of the loop
+***
+<div class = "answer">
+
+According to our code, we know that `j` is the square of `i`, so that's the number we need to print. If we put the `print()` statement **outside** of the loop, we only see the last number, 25, because that is the value that `j` had once the loop was completed. If we place the `print()` statement **inside** the loop, each value of `j` is printed before the next iteration of the loop. Therefore, the second option is correct.
```

Comment:
```suggestion
According to our code, we know that `j` is the square of `i`, so that's the number we need to print. If we put the `print()` statement **outside** of the loop, we only see the last number, 25, because that is the value that `j` had once the loop was completed. If we place the `print(j)` statement **inside** the loop, each value of `j` is printed before the next iteration of the loop. Therefore, the second option is correct.
```

Reply:
Feel free to reject this suggestion, it is slightly clearer to me this way, but might not be to others.

Reply:
I think your way is clearer!